### ISSUE TITLE
error with latest mimemagic version mismatch,  cannot load such file -- mimemagic/overlay 

### ISSUE THREAD

**Issue by oakbow** [carrierwaveuploader/carrierwave#2554]
we need to update mimemagic gem for licensing issue but if you do it you will see the error.

```
../lib/carrierwave/sanitized_file.rb:5:in `require': cannot load such file -- mimemagic/overlay (LoadError)
```

because mimemagic has 2 versions. 0.4.x and 0.3.x.
0.4.x is just a version to remove mimemagic/overlay https://github.com/mimemagicrb/mimemagic/compare/v0.3.7...v0.4.1
but CarrierWave is now depending on this file at `carrierwave/sanitized_file.rb:5`

CarrierWave should depend on 0.3.x.
because of CarrierWave gem dependency setting, you always install mimemagic 0.4.x

https://github.com/carrierwaveuploader/carrierwave/blob/v2.2.0/carrierwave.gemspec#L28

```
  s.add_dependency "image_processing", "~> 1.1"
  s.add_dependency "mimemagic", ">= 0.3.0"  # <<- version should be ~> 0.3.0
  s.add_dependency "addressable", "~> 2.6"
  s.add_dependency "ssrf_filter", "~> 1.0"
```

there are 2 ways to resolve this problem.

1. release new version which contains merged pull request https://github.com/carrierwaveuploader/carrierwave/pull/2551
2. fix the gem dependency above from `>= 0.3.0` to `~> 0.3.0`.

of course I recommend the first.

**Comment by pedrofurtado** [comment]
Any news about this issue? This breaking change in mimemagic are breaking all builds here 😢

**Comment by pedrofurtado** [comment]
pinging some maintainers, to help us 🤝 🍻 : @mshibuya

**Comment by pjmartorell** [comment]
carrierwave master does not depend on mimemagic anymore. For carrierwave 2.2.0 you can circumvent the issue adding mimemagic to your Gemfile but specifiying a version `~> 0.3.10`. Of course it would be better if carrierwave.gemspec could include so is not needed to add it to the Gemfile, releasing a 2.2.1 version with this fix.

**Comment by oakbow** [comment]
there is a resolution by user side.
to avoid this problem users are able to add this line on Gemfile:

```
gem 'mimemagic', '~> 0.3.0'
```

this will be a safety lock to make mimemagic version to be in 0.3.x

**Comment by mshibuya** [comment]
Resolved by #2551.
